src/main/java/github/scarsz/discordsrv/objects/managers/link/JdbcAccountLinkManager.java

```diff
@@ -79,16 +81,25 @@ public static boolean shouldUseJdbc(boolean quiet) {
         if (StringUtils.isBlank(jdbc)) return false;
 
         Matcher matcher = JDBC_PATTERN.matcher(jdbc);
-        if (!matcher.find() || matcher.groupCount() < 4) {
-            if (!quiet) DiscordSRV.error("Not using JDBC because < 4 matches for JDBC url");
+        
+        if (!matcher.matches()) {
+            if (!quiet) DiscordSRV.error("connection string '" + jdbc + "' is NOT valid!");
             return false;
         }
+        
+        if (matcher.group('DBEngine').ToUpper() != 'JDBC')
+        {
+            if (!quiet) DiscordSRV.error("Not using JDBC as DBProtocol in the connection string ");
+            return false;
+        }
```

Comment:
To match the project's styling the first { should be on the same line as the if (). `String#ToUpper` is not a Java method, wrong quotes and Strings in Java should be checked with `equals` instead of `==`, this can also be changed to equalsIgnoreCase instead of making the String uppercase first.
Also changed the error message language

```suggestion
        if (!matcher.group("DBEngine").equalsIgnoreCase("jdbc")) {
            if (!quiet) DiscordSRV.error("Not using JDBC because the protocol of the jdbc url is wrong");
            return false;
        }
```